src/misc/help_util.cpp

```diff
@@ -0,0 +1,52 @@
+#include "dosbox.h"
+
+#include <map>
+#include <string>
+
+#include "help_util.h"
+#include "string_utils.h"
+#include "support.h"
+
+namespace HelpUtil {
+
+static std::map<const std::string, Detail> help_list = {};
+
+void add_to_help_list(const std::string& cmd_name, const Detail& detail, bool replace_existing) {
+    if (replace_existing || !contains(help_list, cmd_name)) {
+        help_list[cmd_name] = detail;
+    }
+}
+
+const std::map<const std::string, Detail>& get_help_list() {
+    return help_list;
+}
+
+std::string get_short_help(const std::string& name) {
+    const std::string short_key = "SHELL_CMD_" + name + "_HELP";
+	if (std::string str = MSG_Get(short_key.c_str()); !starts_with("Message not Found!", str)) {
```

Comment:
Checking for "Message not Found!", I really don't like this... I get it that you want to avoid changing `MSG_Get` because it's used in so many places, but maybe it would be better to introduce a new function that only checks if the message can be found and returns a boolean (and `MSG_Get` could be refactored to internally call that function).

Reply:
Oh, I don't like it either. Good call on creating a new function for it.

Reply:
It will be really nice to eliminate this  "string-as-a-signal" for good.  It was brought into upstream in 2006 from ykhwong, in a56dddbb128c171cf781f47f4bc34d168b513208, and we've been chipping at removing it (some removed in 784c8754a65a5e362849787ce02d2e813c9c7703).  I think this will be the last of it!

Reply:
I added a new `MSG_Exists()` function to test if a message key exists.